packages/spanner/src/database.js

```diff
@@ -577,6 +593,25 @@ Database.prototype.run = function(query, options, callback) {
  *   .on('end', function() {});
  *
  * //-
+ * // If you need to enforce a specific param type, a types map can be provided.
+ * // This is typically useful if your param value can be null.
```

Comment:
Is it just this one case, where the user provides `null` and we have no way to know what the value's non-null type is? Or are there others as well?

This being the solution for the null thing makes sense to me. The only change I would suggest is, if that *is* the only case, we should update the example to show the value passed in as `name` being `null`. Otherwise, at a glance, it seems like we force explicit listing of parameter types, when we actually get it right most of the time.